### PR TITLE
Add background tasks to register endpoint and on_after_register handler

### PR DESCRIPTION
**Pull request by jankadel**
It would be great to have background tasks available in after register handler.

### CODE REVIEW

File: fastapi_users/manager.py [hunk 1]
```diff
@@ -110,6 +110,7 @@ async def get_by_oauth_account(self, oauth: str, account_id: str) -> models.UP:
     async def create(
         self,
         user_create: schemas.UC,
+        background_tasks: BackgroundTasks,
         safe: bool = False,
         request: Optional[Request] = None,
```
Comment (sorXCode) [inline fastapi_users/manager.py:115]: Adding the `background_tasks` as the last and optional function parameter.

```suggestion
        safe: bool = False,
        request: Optional[Request] = None,
        background_tasks: Optional[BackgroundTasks] = None,
```

File: fastapi_users/manager.py [hunk 2]
```diff
@@ -498,7 +499,7 @@ async def validate_password(
         return  # pragma: no cover
 
     async def on_after_register(
-        self, user: models.UP, request: Optional[Request] = None
+        self, user: models.UP, request: Optional[Request] = None, background_tasks: BackgroundTasks | None = None
```
Comment (sorXCode) [inline fastapi_users/manager.py:502]:
```suggestion
        self,
        user: models.UP,
        request: Optional[Request] = None,
        background_tasks: Optional[BackgroundTasks] = None,
```

File: fastapi_users/router/register.py [hunk 3]
```diff
@@ -73,4 +74,4 @@ async def register(
 
         return schemas.model_validate(user_schema, created_user)
 
-    return router
+    return router
```
Comment (sorXCode) [inline fastapi_users/router/register.py:77]:
```suggestion
    return router

```

**Comment by frankie567** [comment]
Thank you for the idea and proposal @jankadel! However, you can achieve this by adding `background_tasks` to the `UserManager` initializer. For example:

```py
class UserManager(UUIDIDMixin, BaseUserManager[User, uuid.UUID]):
    reset_password_token_secret = SECRET
    verification_token_secret = SECRET

    def __init__(self, user_db: SQLAlchemyUserDatabase, background_tasks: BackgroundTasks):
        super().__init__(user_db)
        self.background_tasks = background_tasks
        
    async def on_after_register(self, user: User, request: Optional[Request] = None):
        print(f"User {user.id} has registered.")
        self.background_tasks.add_task(my_task, user=user)

async def get_user_manager(background_tasks: BackgroundTasks, user_db=Depends(get_user_db)):
    yield UserManager(user_db, background_tasks)
```

So, there is no need to add another param to the hooks 🙂

**Comment by jankadel** [comment]
Ah great! Thanks @frankie567

**Pull request closed without merging**